src/gui/transferlistwidget.cpp

```diff
@@ -428,65 +428,56 @@ void TransferListWidget::previewSelectedTorrents()
 
 void TransferListWidget::setDlLimitSelectedTorrents()
 {
-    QList<BitTorrent::TorrentHandle *> selected_torrents;
-    bool first = true;
-    bool all_same_limit = true;
+    QList<BitTorrent::TorrentHandle *> TorrentsList;
     foreach (BitTorrent::TorrentHandle *const torrent, getSelectedTorrents()) {
-        if (!torrent->isSeed()) {
-            selected_torrents << torrent;
-            // Determine current limit for selected torrents
-            if (first)
-                first = false;
-            else if (all_same_limit && (torrent->downloadLimit() != selected_torrents.first()->downloadLimit()))
-                all_same_limit = false;
-        }
+        if (torrent->isSeed())
+            continue;
+        TorrentsList += torrent;
     }
+    if (TorrentsList.empty()) return;
 
-    if (selected_torrents.empty()) return;
+    int oldLimit = TorrentsList.first()->downloadLimit();
+    foreach (BitTorrent::TorrentHandle *const torrent, TorrentsList) {
```

Comment:
>Answer for the second question: no, not even for 1000000 items, ugh I mean yes, but it is barely noticeable even for 1000000 items, run time diff could be less than 1 sec.
Then the first question: it depends on how much you want to optimize every bit out of it, for me, I have no problem converting it over.

LOL. I would be surprised if qbt was still usable with that many torrents.
So QList is better to keep the "qt spirit" :)

>Last, I've an idea to make it a bit more concise, let me code it later.

ok